File: backend/rpc/handler/hello/hello.py
from rpc.gen.hello.hello.services import THelloService
from rpc.gen.user.user.types.ttypes import TUserRole

from models import models

class THelloServiceHandler(THelloService.Iface):
    def __init__(self):
        self.auth_model = models['auth']

    def hello_admin_utama(self, auth_token):
        auth_payload = self.auth_model.require_role(auth_token, TUserRole.ADMIN_UTAMA)
        return "Halo, Admin Utama!"

File: frontend/src/rpc/client/hello/hello.js
import THelloService from '@/rpc/gen/THelloService';
import { TBaseClient } from '@/rpc/client/base';
import { TUserRole } from '@/rpc/gen/user.user.types_types';


class THelloClient extends TBaseClient{
	constructor(stores=null){
		super(stores, THelloService, '/api/hello/hello');
		return this;
	}

	async hello_admin_utama(){
		this.stores.helper.auth.requireRole(TUserRole.ADMIN_UTAMA);
		return await this.client.hello_admin_utama(this.authStore.authToken);
	}

}

const helloClient = new THelloClient();

export { THelloClient, helloClient }
export default helloClient